src/managed_data/_api.py

```diff
@@ -1,23 +1,47 @@
-"""The `deeporigin.managed_data._api` module contains low-level functions for interacting
-with Deep Origin's managed data API. The functions in this module
-simply provide Pythonic interfaces to individual API endpoints."""
-
-import os
-from typing import Optional, Union
-from urllib.parse import parse_qs, urlparse
-
-from beartype import beartype
-from deeporigin.exceptions import DeepOriginException
-from deeporigin.managed_data.client import Client, DeepOriginClient
-from deeporigin.managed_data.schema import (
-    Cardinality,
-    DataType,
-    RowType,
-)
-
-
-@beartype
-def _get_default_client(client: Optional[Client] = None) -> Client:
+import inspect
+import sys
+
+from deeporigin import auth
+from deeporigin.utils import _get_method
+from deeporigin_data import DeeporiginData
+
+# this dict allows us to simply wrap endpoints in the low-level
+# API so that we can more easily call them, without needing
+# to explicitly write functions for them
```

Comment:
This approach feels a bit brittle and strange, but I don't have the context of _why_. Perhaps worth explaining why you took this path in a code comment would be useful.